app/controllers/download_controller.rb

```diff
@@ -65,4 +64,20 @@ def download_user_params
       :register
     )
   end
+
+  def download_from_uri(uri)
+    headers = {}
+    headers['Authorization'] = api_key if api_key.present?
+
+    response = HTTParty.get(uri, headers: headers)
+    if response.code == 200
+      response.body
+    else
+      raise "#{response.status}: #{response.message}"
```

Comment:
will this return a 500 to the end user? Maybe we can create our own exception class to better capture when this happens?

Reply:
Yes this will return a 500 page. I'm not sure what you mean by "better capture" - do you want to show something different to the user?

Reply:
No, I just mean if we create an exception class we will see a consistent exception class in the logs we can better query for.